src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -154,171 +155,179 @@ private RootCommand SetupCommandLineAPI()
+            FinishAnalysis();
+            
+            return (int)exitCode;
+        }
+
+        private ExitCode AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath)
+        { 
+            try
             {
-                if (disposeWriter && writer != null)
-                {
-                    writer.Dispose();
+                string templateFileContents = File.ReadAllText(templateFilePath.FullName);
+                string parameterFileContents = parametersFilePath == null ? null : File.ReadAllText(parametersFilePath.FullName);
 
-                    // This block will only be executed if the CLI was called with analyze-template
-                    this.summaryLoggerProvider.SummaryLogger.SummarizeLogs(verbose);
-                }
+                IEnumerable<IEvaluation> evaluations = this.templateAnalyzer.AnalyzeTemplate(templateFileContents, parameterFileContents, templateFilePath.FullName);
+
+                this.reportWriter.WriteResults(evaluations, (FileInfoBase)templateFilePath, (FileInfoBase)parametersFilePath);
+
+                return evaluations.Any(e => !e.Passed) ? ExitCode.Violation : ExitCode.Success;
+            }
+            catch (Exception exception)
+            {
+                logger.LogError(exception, "An exception occurred while analyzing a template");
+                return ExitCode.ErrorGeneric;
             }
         }
 
-        private int AnalyzeDirectory(DirectoryInfo directoryPath, FileInfo configurationsFilePath, ReportFormat reportFormat, FileInfo outputFilePath, bool runTtk, bool verbose)
+        private ExitCode SetupAnalysis(
+            FileInfo configurationFile,
+            DirectoryInfo directoryToAnalyze,
+            ReportFormat reportFormat,
+            FileInfo outputFilePath,
+            bool runPowershell,
+            bool verbose)
         {
-            // Check that output file path provided for sarif report
+            // Output file path must be specified if SARIF was chosen as the report format
             if (reportFormat == ReportFormat.Sarif && outputFilePath == null)
             {
-                // We can't use the logger for this error,
-                // because we need to get the writer to create the logger,
-                // but this check has to be done before getting the writer:
-                Console.WriteLine("When using --report-format sarif flag, --output-file-path flag is required.");
-                return (int)ExitCode.ErrorMissingPath;
+                Console.Error.WriteLine("When using --report-format sarif flag, --output-file-path flag is required.");
+                return ExitCode.ErrorMissingPath;
             }
 
-            using var reportWriter = GetReportWriter(reportFormat, outputFilePath, directoryPath.FullName);
+            this.reportWriter = GetReportWriter(reportFormat, outputFilePath, directoryToAnalyze?.FullName);
+            CreateLoggers(verbose);
```

Comment:
Why is it plural now when it's only creating one logger?

Reply:
I think of it as creating multiple:
1. Console logger
2. Summary logger
3. Maybe: Sarif logger